cli/txctx/tx.go

```diff
@@ -68,12 +77,32 @@ func SignAndSend(ctx *cli.Context, act *actor.Actor, acc *wallet.Account, tx *tr
 			// Compensate for confirmation waiting.
 			tx.ValidUntilBlock += uint32((waitTime.Milliseconds() / int64(ver.Protocol.MillisecondsPerBlock))) + 1
 		}
-		_, _, err = act.SignAndSend(tx)
+		resTx, vub, err := act.SignAndSend(tx)
+		if err != nil {
+			return cli.NewExitError(err, 1)
+		}
+
+		if ctx.Bool("await") {
+			aer, err = act.Wait(resTx, vub, err)
+		}
 	}
 	if err != nil {
 		return cli.NewExitError(err, 1)
 	}
 
-	fmt.Fprintln(ctx.App.Writer, tx.Hash().StringLE())
+	DumpTransactionInfo(ctx.App.Writer, tx.Hash(), aer)
 	return nil
 }
+
+// DumpTransactionInfo prints transaction info to the given writer.
+func DumpTransactionInfo(w io.Writer, h util.Uint256, res *state.AppExecResult) {
+	fmt.Fprintln(w, h.StringLE())
+	if res != nil {
+		fmt.Fprintf(w, "OnChain:\t%t\n", res != nil)
+		fmt.Fprintf(w, "BlockHash:\t%s\n", res.Container.StringLE())
```

Comment:
Also my bad, remove BlockHash line, because res.Container is transaction hash (container of the transaction script execution).